Source/UI/NewUI/PaymentForm/View/PaymentHeaderView.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  PaymentHeaderView.swift
+//  Frames
+//
+//  Created by Ehab Alsharkawy
+//  Copyright © 2022 Checkout. All rights reserved.
+//
+
+import UIKit
+
+public final class PaymentHeaderView: UIView {
+  private var style: PaymentHeaderCellStyle?
+
+  private var stackView: UIStackView = {
+    let view = UIStackView().disabledAutoresizingIntoConstraints()
+    view.axis = .vertical
+    view.spacing = 10
+    return view
+  }()
+
+  private lazy var headerLabel: LabelView = {
+    LabelView()
+  }()
+
+  private lazy var subtitleLabel: LabelView = {
+    LabelView()
+  }()
+
+  private(set) lazy var iconsViewContainer: UIView = {
+    UIView()
+  }()
+
+  private(set) lazy var iconsStackView: UIStackView = {
+    let view = UIStackView().disabledAutoresizingIntoConstraints()
+    view.axis = .horizontal
+    view.alignment = .leading
+    view.distribution = .fillEqually
+    view.spacing = Constants.Padding.xs.rawValue
+    return view
+  }()
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    setupViewsInOrder()
+  }
+
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  func update(style: PaymentHeaderCellStyle?) {
+    guard let style = style else { return }
+    self.style = style
+
+    backgroundColor = style.backgroundColor
+    headerLabel.isHidden = true
+    subtitleLabel.isHidden = true
+
+    if let headerStyle = style.headerLabel {
+      headerLabel.isHidden = false
+      headerLabel.update(with: headerStyle)
+    }
+    if let subtitleStyle = style.subtitleLabel {
+      subtitleLabel.isHidden = false
+      subtitleLabel.update(with: subtitleStyle)
+    }
+
+    addArrangedSubviewForStackView()
+  }
+
+  private func addArrangedSubviewForStackView() {
+    guard let icons = style?.schemaIcons else {
+      iconsStackView.isHidden = true
+      return
+    }
+    iconsStackView.isHidden = false
+    for image in icons where image != nil {
+      let imageView = UIImageView().disabledAutoresizingIntoConstraints()
+      imageView.image = image?.withRenderingMode(.alwaysOriginal)
+      imageView.contentMode = .scaleAspectFit
+      imageView.heightAnchor.constraint(equalToConstant: 20).isActive = true
+      imageView.widthAnchor.constraint(equalToConstant: 33).isActive = true
+      iconsStackView.addArrangedSubview(imageView)
+    }
+  }
```

Comment:
```swift
private func updateIconsStack(images: SchemaIcons) {
  iconsStackView.isHidden = images.isEmpty
    
  // Option 1 with bad animation effect if there is an update
  iconsStackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
  // Add images as previously

  // Option 2, if we expect updates during session, make a diff between what is already present and what's new, ensuring smoother UI
  // Track already displayed icons
  let existingIconImageViews = iconsStackView.arrangedSubviews.compactMap({ $0 as? UIImageView })
    
  // Remove any old images that are no longer needed
  existingIconImageViews.forEach { 
    guard image = $0.image,
      images.contains(image) else {
        $0.removeFromSuperview() // No image found in ImageView, or image is no longer needed
        return
    }
  }

  let existingImages = existingIconImageViews.compactMap({ $0.image })
  for image in images where !existingImages.contains(image) {
    // Add image as previously
  }
}
```

Reply:
Icons will only updated once, no need for removing anyone of them.